core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/LruBlockCache.java

```diff
@@ -222,6 +226,10 @@ public LruBlockCache(long maxSize, long blockSize, boolean evictionThread, int m
     this.overhead = calculateOverhead(maxSize, blockSize, mapConcurrencyLevel);
     this.size = new AtomicLong(this.overhead);
 
+    for (int i = 0; i < loadLocks.length; i++) {
+      loadLocks[i] = new ReentrantLock(false);
```

Comment:
Is LruBlockCache used like a singleton? If not, should loadLocks be static? I'm just thinking we probably don't want 5000 objects created every time we create an LruBlockCache.

Reply:
In the tablet server only a single instance is created, but its not a singleton.  However in the RFile API code an instance is created per scanner AND that code is not thread safe, so it may make sense to disable creating the locks there.

Reply:
> If not, should loadLocks be static

Would not want it to be static.  This could cause an index cache operation to block on a data cache operation.

Reply:
I disabled creating locks in the RFile client code.